src/rust/bitbox02-rust/src/platform/bitboxbase/config.rs

```diff
@@ -0,0 +1,163 @@
+// Copyright 2019 Shift Cryptosecurity AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use core::fmt::Write;
+use core::time::Duration;
+
+use arrayvec::ArrayString;
+
+use crate::c_char;
+use crate::error::Error;
+use crate::util::{FCString, Ipv4Addr};
+
+#[derive(Debug, Clone)]
+#[repr(u8)]
+#[allow(dead_code)] // We are settings this with transmute
+pub enum StatusLedMode {
+    Always,
+    OnWorking,
+    OnWarning,
+    OnError,
+}
+
+#[derive(Debug, Clone)]
+#[repr(u8)]
+#[allow(dead_code)] // We are settings this with transmute
+pub enum StatusScreenMode {
+    Always,
+    OnWorking,
+    OnWarning,
+    OnError,
+}
+
+pub struct Config {
+    pub(crate) status_led_mode: StatusLedMode,
+    pub(crate) status_screen_mode: StatusScreenMode,
+    pub(crate) default_display_duration: Duration,
+    pub(crate) hostname: Option<ArrayString<[u8; 64]>>,
+    pub(crate) ip: Option<Ipv4Addr>,
+}
+
+impl Config {
+    pub const fn new() -> Config {
+        Config {
+            status_led_mode: StatusLedMode::Always,
+            status_screen_mode: StatusScreenMode::OnWarning,
+            default_display_duration: Duration::from_secs(10),
+            hostname: None,
+            ip: None,
+        }
+    }
+
+    pub fn set_status_led_mode(&mut self, mode: StatusLedMode) {
+        self.status_led_mode = mode;
+    }
+
+    pub fn set_status_screen_mode(&mut self, mode: StatusScreenMode) {
+        self.status_screen_mode = mode;
+    }
+
+    pub fn set_hostname(&mut self, hostname: &str) -> Result<(), Error> {
+        for c in hostname.chars() {
+            if !c.is_alphanumeric() {
+                print_debug!(5000, "Invalid char: {}", c);
+            }
+        }
+        if hostname.chars().any(|c| !c.is_alphanumeric()) {
```

Comment:
same check as above, refactor to use one source for checking. could also just drop the debug print above.

Reply:
ack